src/main/java/ch/uzh/ifi/seal/soprafs20/service/GameService.java

```diff
@@ -86,6 +89,28 @@ public Long createGame(List<Long> players) {
         return gameId;
     }
 
+    public void submitWord(long id, String word) {
+        WordCheck wordChecker = new WordCheck();
+        Game game = gameRepository.findById(id)
+            .orElseThrow(
+                () -> new NotFoundException(String.format("A game with the id %id was not found", id))
+            );
+        if (!wordChecker.checkEnglishWord(word)) {
+            throw new ServiceException("The clue submitted is not an English word");
+        }
+        List<String> clues = game.getClues();
+        if (clues.size() >= game.getPlayerIds().size() - 1) {
```

Comment:
Please check this condition. The condition should be greater than only. If there are 5 players, then this condition will start throwing exception for the 4th player.

Reply:
I don't think so, lets say we have the last player submitting the clue, then the clue size in that line is getPlayerIds().size() - 2 . The clue only gets added to the list after the check.

Reply:
oh right..my bad. You are setting the clue after checking this.